src/ray/core_worker/core_worker.cc

```diff
@@ -238,6 +249,53 @@ void CoreWorkerProcess::RunTaskExecutionLoop() {
   instance_.reset();
 }
 
+void CoreWorkerProcess::RunStatsService() {
+  absl::MutexLock lock(&CoreWorkerProcess::stats_initialization_mutex_);
+  enable_stats_count_++;
+  // Assume stats module will be initialized exactly once in once process.
+  if (stats_thread_) {
+    RAY_LOG(DEBUG) << "Stats module has been initialized and it does not need to "
+                   << "setup twice.";
+    return;
+  }
+  RAY_LOG(DEBUG) << "Stats setup.";
+  // Initialize stats in core worker global tags.
+  const ray::stats::TagsType global_tags = {{ray::stats::ComponentKey, "core_worker"},
+                                            {ray::stats::VersionKey, "0.9.0.dev0"}};
+
+  // NOTE(lingxuan.zlx): We assume RayConfig is initialized before it's used.
+  // RayConfig is generated in Java_io_ray_runtime_RayNativeRuntime_nativeInitialize
+  // for java worker or in constructor of CoreWorker for python worker.
+  ray::stats::Init(global_tags, RayConfig::instance().metrics_agent_port(),
+                   CoreWorkerProcess::stats_io_service_);
+  CoreWorkerProcess::stats_thread_.reset(
+      new std::thread([]() { CoreWorkerProcess::stats_io_service_.run(); }));
+  // NOTE(lingxuan.zlx): Should shutdown if opencensus has stats shutdown
+  // api. But currently no such shutdown api has been provided in opencensus.
+  // For sake of running without crashed we need detach the stats thread if
+  // it stands for a driver. Besides, detached exporter thread might be halted
+  // when process has been terminated without side effects.
+  // CoreWorkerProcess::stats_thread_->detach();
+}
+
+void CoreWorkerProcess::StopStatsService() {
+  absl::MutexLock lock(&CoreWorkerProcess::stats_initialization_mutex_);
+  RAY_LOG(DEBUG) << "Stats stop.";
+  if (!stats_thread_) {
+    return;
+  }
+  enable_stats_count_--;
+  if (enable_stats_count_ > 0) {
+    return;
+  }
```

Comment:
Please leave a comment about this - I assume it's handling the multi-worker case?

Reply:
> Please leave a comment about this - I assume it's handling the multi-worker case?

The parameter `enable_stats_count_ ` stands for how many core workers are initialized without stats module, so we can setup stats metric exporter once and shutdown at the end of last core work lifecycle.